src/libraries/System.Net.Security/src/System/Net/Security/SslCertificateTrust.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Security
+{
+    public sealed class SslCertificateTrust
+    {
+        internal X509Store? _store;
+        internal X509Certificate2Collection? _trustList;
+        internal bool _sendTrustInHandshake;
+
+        public static SslCertificateTrust CreateForX509Store(X509Store store, bool sendTrustInHandshake = false)
+        {
+
+#if TARGET_WINDOWS
```

Comment:
OperatingSystem.IsWindows is runtime check. This may not matter much but with the `#if` the code would not be event there on other platforms. We may end-up with platform files like we do for `SslStreamCertificateContext` and that would achieve similar goal. I'll take another look when adding Linux and macOS support.